meson.build

```diff
@@ -0,0 +1,25 @@
+project('shtools', 'c', 'fortran',
+  version: '4.11.0dev0',
+  license: 'BSD-3-Clause',
```

Comment:
The meson and meson-python docs are a little confusing and inconsistent as to what the default buildtype and optimization level is. meson claims that the default buildtype is debug optimize and meson-python claims it is release. meson-python claims that the default optimization is 2, but when I checked `meson configure`, it said that we were using level 3. It might be prudent to set the following here:

```
default_options: [
  'buildtype=release',
  'optimization=3',
]
```

Reply:
This is because meson-python will by default act in the same position as an end user, and pass configurations to meson to use what meson considers a non-default build type.